vault/router.go

```diff
@@ -347,9 +347,15 @@ func (r *Router) routeCommon(req *logical.Request, existenceCheck bool) (*logica
 	originalClientTokenRemainingUses := req.ClientTokenRemainingUses
 	req.ClientTokenRemainingUses = 0
 
-	// Cache the headers and hide them from backends
+	// Cache the headers and hide all but a small, standard set of them from backends
```

Comment:
I'm not sure an administrator view is that useful.  The headers (and indeed the body) which backends need (this being one of two that need access to the underlying request) is really a function of the backend, rather than the administrator - e.g. it wouldn't seem sensible to allow an administrator to disable a header required by a backend.

So perhaps the list of headers (and whether their values should be hashed in the audit logs) should be requested by the backend at mount time?